packages/core/src/ctype/CType.utils.ts

```diff
@@ -77,28 +78,34 @@ export async function verifyOwner(ctype: ICType): Promise<boolean> {
   return owner ? owner === ctype.owner : false
 }
 
-type schemaPropsForHashing = {
-  $schema: ICType['schema']['$schema']
-  properties: ICType['schema']['properties']
-  title: ICType['schema']['title']
-  type: ICType['schema']['type']
+export function getSchemaPropertiesForHash(
+  ctypeSchema: CTypeSchemaWithoutId | ICType['schema']
+): Partial<ICType['schema']> {
+  // We need to remove the CType ID from the CType before storing it on the blockchain
+  // otherwise the resulting hash will be different, as the hash on chain would contain the CType ID,
+  // which is itself a hash of the CType schema.
+  if ((ctypeSchema as ICType['schema']).$id) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { $id, ...schemaWithoutId } = ctypeSchema as ICType['schema']
```

Comment:
W: I personally would prefer making a full copy and deleting the $id (with `delete` command), instead of having an unused variable lying around.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/415/commits/8f28266f9af709418933b78a015c586e5695bba2